chandra_aca/dark_subtract.py

```diff
@@ -0,0 +1,186 @@
+import numba
+import numpy as np
+import requests.exceptions
+import scipy.signal
+from cheta import fetch_sci
+from ska_helpers import retry
+from ska_numpy import smooth
+
+from chandra_aca.dark_model import dark_temp_scale_img
+
+
+@retry.retry(exceptions=requests.exceptions.RequestException, delay=5, tries=3)
+def get_tccd_data(
+    times, smooth_window=30, median_window=3, source="maude", maude_channel=None
+):
+    """
+    Get the CCD temperature for given times and interpolate and smooth.
+
+    This is a light wrapper around cheta.fetch_sci.Msid("aacccdpt", start, stop)
+    to handle an option to use the maude data source in an explicit way.
+
+    Parameters
+    ----------
+    times: np.array float of Chandra seconds
+        The times for which to get the CCD temperature data.
+    source : str, optional
+        Source of the CCD temperature data. If 'maude', override the fetch_sci data source
+        to 'maude allow_subset=False'. Else, use the cheta default.
+    median_window : int, optional
+        3-sample Median filter to to remove outliers.
+    smooth_window : int, optional
+        Smooth the data using a hanning window of this length in samples.
+
+
+    Returns
+    -------
+    vals : np.array
+        CCD temperature values.
+    """
+
+    pad = 600  # 600 seconds default pad
+
+    # increase the pad if the smooth window is large
+    pad = np.max([smooth_window * 32.8 / 2, pad])
+
+    fetch_start = times[0] - pad
+    fetch_stop = times[-1] + pad
+
+    if source == "maude":
+        # Override the cheta data_source to be explicit about maude source.
+        data_source = "maude allow_subset=False"
+        if maude_channel is not None:
+            data_source += f" channel={maude_channel}"
+        with fetch_sci.data_source(data_source):
+            dat = fetch_sci.Msid("aacccdpt", fetch_start, fetch_stop)
+    else:
+        dat = fetch_sci.Msid("aacccdpt", fetch_start, fetch_stop)
+
+    yin = dat.vals.copy()
+
+    if median_window > 0:
+        # Filter the data using a 3 point median filter from scipy
+        yin = scipy.signal.medfilt(yin, kernel_size=median_window)
+
+    if smooth_window > 0:
+        # And then smooth it with hanning and window_len = smooth_window
+        yin = smooth(yin, window_len=smooth_window)
+
+    # Interpolate the data to the requested times
+    vals = np.interp(times, dat.times, yin)
+
+    return vals
+
+
+def get_aca_images_bgd_sub(img_table, t_ccd_vals, img_dark, tccd_dark):
```

Comment:
I think this is close to matching what you drafted as a new signature in https://github.com/sot/chandra_aca/pull/174#discussion_r1822440753 .